### PR TITLE
RSE-32: Create Case Type Category field

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR, the Case Type Category field is added to Civicase.

## Before
civicrm_case_type-before

## After
Field is Added
2019-07-01 at 4 49 PM

Option Group is created
2019-07-01 at 4 50 PM

## Technical Details
**Upgrader**
Support for adding upgraders in individual files has been added.

A new upgrader has been added which adds a new field `case_type_category` to `civicrm_case_type` table. 
Also it adds a new Option Group called `Case Type Categories`.

**hook_civicrm_entityTypes**
A new function `_civicase_update_case_type_entity` is added, which adds the `case_type_category` field to the `CaseType` DAO.

## Comments
`civilint` fixes has been made to the touched files. But there will be a discussion with Davi regarding this, and once finalised, `civilint`(or any other linter) will be added as part of pre commit hooks.

### CODE REVIEW

File: CRM/Civicase/Upgrader.php [hunk 1]
```diff
@@ -408,66 +414,19 @@ public function upgrade_0001() {
...
-   *
-  public function upgrade_4200() {
-    $this->ctx->log->info('Applying update 4200');
-    CRM_Core_DAO::executeQuery('UPDATE foo SET bar = "whiz"');
-    CRM_Core_DAO::executeQuery('DELETE FROM bang WHERE willy = wonka(2)');
-    return TRUE;
-  } // */
+   * Adds "Case Type Categories" field to the `civicrm_case_type` table
+   */
+  public function upgrade_0002() {
```
Comment (tunbola) [inline CRM/Civicase/Upgrader.php]: Can we follow the upgrader process for adding a Civicrm upgrader described here: https://compucorp.atlassian.net/wiki/spaces/PCHR/pages/835092529/Add+a+CiviCRM+Upgrader.

File: civicase.php [hunk 2]
```diff
@@ -637,6 +622,8 @@ function civicase_civicrm_entityTypes(&$entityTypes) {
     'class' => 'CRM_Civicase_DAO_CaseContactLock',
     'table' => 'civicase_contactlock',
   );
+
+  _civicase_update_case_type_entity($entityTypes);
```
Comment (tunbola) [inline civicase.php]: Why don't we give this a more descriptive name e.g `_civicase_add_case_category_case_type_entity`

File: CRM/Civicase/Upgrader.php (hunk 1 above)
Comment (deb1990) [reply CRM/Civicase/Upgrader.php]: Fixed.

File: CRM/Civicase/Upgrader.php [hunk 3]
```diff
@@ -405,28 +405,94 @@ private function getCaseMenuItem($name) {
...
+    }
+    ksort($sortedKeyedClasses, SORT_NUMERIC);
+    return $sortedKeyedClasses;
+  }
 
-    return TRUE;
+  /**
+   * Gets the PEAR style classname from an upgrader file
+   *
+   * @param $file
```
Comment (tunbola) [inline CRM/Civicase/Upgrader.php]: missing param type.

**Pull request merged**